Structure.Rmd

```diff
@@ -0,0 +1,398 @@
+  typically files that facilitate your development process, but that should be
+  excluded from the distributed product.
+
+#### `.Rbuildignore` {#rbuildignore}
+
+You won't need to contemplate the exact structure of package `.tar.gz` files very often, but you do need to understand the `.Rbuildignore` file. It controls which files from the source package make it into the downstream forms.
+
+Each line of `.Rbuildignore` is a Perl-compatible regular expression that is matched, without regard to case, against the path to each file in the source package, i.e. the result of `dir(full.names = TRUE, recursive = TRUE, include.dirs = TRUE, all.files = TRUE)` run from the package's top-level directory. If the regular expression matches, that file or directory is excluded. Note there are some default exclusions implemented by R itself, mostly relating to classic version control systems and editors, such as SVN, Git, and Emacs.
+
+To exclude a specific file or directory (the most common use case), you __MUST__ anchor the regular expression. For example, to exclude a directory called "notes", use `^notes$`. The regular expression `notes` will match any file name containing notes, e.g. `R/notes.R`, `man/important-notes.R`, `data/endnotes.Rdata`, etc. The safest way to exclude a specific file or directory is to use `usethis::use_build_ignore("notes")`, which does the escaping for you.
+
+`.Rbuildignore` is a way to resolve some of the tension between the practices that support your development process and CRAN's requirements for submission and distribution. The affected files fall into two broad, semi-overlapping classes:
+
+  * Files that help you generate package contents programmatically. Examples:
+    - Using `README.Rmd` to generate an informative and current `README.md`.
+    - Storing `.R` scripts to create and update internal or exported data.
+  * Files that drive package development, checking, and documentation, outside
+    of CRAN's purview. Examples:
+    - Files relating to the RStudio IDE.
+    - Using the [pkgdown package](https://pkgdown.r-lib.org) to generate a
+      website.
+    - Configuration files related to continuous integration/deployment and 
+      monitoring test coverage.
+
+Here is a non-exhaustive list of typical entries in the `.Rbuildignore` file for a package in the tidyverse:
+
+```
+^.*\.Rproj$         # Designates the directory as an RStudio Project
+^\.Rproj\.user$     # Used by RStudio for temporary files
+^README\.Rmd$       # An Rmd file used to generate README.md
+^LICENSE\.md$       # Full text of the license
+^cran-comments\.md$ # Comments for CRAN submission
+^\.travis\.yml$     # Used by Travis-CI for continuous integration testing
+^data-raw$          # Code used to create data included in the package
+^pkgdown$           # Resources used for the package website
+^_pkgdown\.yml$     # Configuration info for the package website
+^\.github$          # Contributing guidelines, CoC, issue templates, etc.
+```
```

Comment:
This section saw quite a bit of revision because the world and our practices have changed.